genie/validate.py

```diff
@@ -12,31 +12,108 @@
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
+class Validator(object):
+    """A validator helper class for a center's files.
+    """
 
-def determine_filetype(syn, filepathlist, center):
-    '''
-    Get the file type of the file by validating its filename
-
-    Args:
-        syn: Synapse object
-        filepathlist: list of filepaths to center files
-        center: Participating Center
-
-    Returns:
-        str: File type of input files.  None if no filetype found
-    '''
-    filetype = None
-    # Loop through file formats
-    for file_format in PROCESS_FILES:
-        validator = PROCESS_FILES[file_format](syn, center)
-        try:
-            filetype = validator.validateFilename(filepathlist)
-        except AssertionError:
-            continue
-        # If valid filename, return file type.
-        if filetype is not None:
-            break
-    return(filetype)
+    def __init__(self, syn, center, filepathlist, format_registry=PROCESS_FILES):
+        """A validator helper class for a center's files.
+
+        Args:
+            center: Participating Center
+        """
+
+        self._synapse_client = syn
+        self.filepathlist = filepathlist
+        self.center = center
+        self._format_registry = format_registry
+        self.file_type = self.determine_filetype()
+
+    def determine_filetype(self):
+        '''
+        Get the file type of the file by validating its filename
+
+        Args:
+            syn: Synapse object
+            filepathlist: list of filepaths to center files
+
+        Returns:
+            str: File type of input files.  None if no filetype found
+        '''
+        filetype = None
+        # Loop through file formats
+        for file_format in self._format_registry:
+            validator = self._format_registry[file_format](self._synapse_client, self.center)
+            try:
+                filetype = validator.validateFilename(self.filepathlist)
```

Comment:
`example_file_format.validateFilename` should be something like `example_file_format.validate_file_type`.

Reply:
This is a note only - requires bigger refactor. #185